In [39]:
import math
from scipy import integrate

def dNdE_powerLaw(E, N0, G): # norm, index, energy (TeV)
    dNdE = N0*math.exp(-G*E)
    return dNdE
    

$$\frac{dN}{dE}=N_0 e^{-\Gamma E}$$

In [40]:
# set the fit parameters for each source 
norm_SgrA = 6.113e-8
index_SgrA = 2.57
norm_Crab = 3.157e-7 
index_Crab = 2.525 

Emin_Crab = 0.5 # TeV 
Emin_SgrA = 2.0 # TeV
Emax_both = 40. # TeV

gamma_rate_SgrA = 0.155 # gamma / min 
gamma_rate_Crab = 2.63  # gamma / min

In [43]:
# integrate the differential energy distribution for Sgr A* 
int_tuple_SgrA = integrate.quad(dNdE_powerLaw, Emin_SgrA, math.inf, args=(norm_SgrA,index_SgrA))
int_flux_SgrA = int_tuple_SgrA[0]
int_upperBoundErr_SgrA = int_tuple_SgrA[1]

# now for the Crab with the same bounds as for Sgr A*
int_tuple_Crab_1 = integrate.quad(dNdE_powerLaw, Emin_SgrA, math.inf, args=(norm_Crab,index_Crab))
int_flux_Crab_1 = int_tuple_Crab_1[0]
int_upperBoundErr_Crab_1 = int_tuple_Crab_1[1]

# do the Crab with its full, larger energy range
int_tuple_Crab_2 = integrate.quad(dNdE_powerLaw, Emin_Crab, math.inf, args=(norm_Crab,index_Crab))
int_flux_Crab_2 = int_tuple_Crab_2[0]
int_upperBoundErr_Crab_2 = int_tuple_Crab_2[1]

print(int_flux_SgrA)
print(int_flux_Crab_1)
print(int_flux_Crab_2)

1.3933097795336147e-10
8.013571270764387e-10
3.537662956185068e-08


In [44]:
print("The ratio of Sgr A* flux to the Crab flux, integrated from 2TeV to 40TeV for both:")
print(int_flux_SgrA/int_flux_Crab_1)

print("The ratio of Sgr A* flux to the Crab flux, integrated from 0.5TeV to 40TeV for Crab")
print(int_flux_SgrA/int_flux_Crab_2)

print("The ratio of Sgr A* gamma rate to Crab gamma rate, as reported by VEGAS")
print(gamma_rate_SgrA/gamma_rate_Crab)

The ratio of Sgr A* flux to the Crab flux, integrated from 2TeV to 40TeV for both:
0.1738687699224408
The ratio of Sgr A* flux to the Crab flux, integrated from 0.5TeV to 40TeV for Crab
0.003938503460590059
The ratio of Sgr A* gamma rate to Crab gamma rate, as reported by VEGAS
0.05893536121673004


<table border="1">
<tr>
<th>Source</th>
<th>Flux from 2TeV to Infiniti</th>
</tr>
<tr>
<td>Sgr A*</td>
<td>$$1.393\times10^{-10}$$</td>
</tr>
<tr>
<td>Crab</td>
<td>$$8.01\times10^{-10}$$</td>
</tr>
<tr>
<td>Ratio</td>
<td>$$0.1739$$</td>
</tr>
</table>

$$Flux_{SgrA}={{a}}\frac{gamma}{{m^2s}}$$